src/haystack/val/number.rs

```diff
@@ -126,21 +126,31 @@ impl Hash for Number {
 
 impl PartialEq for Number {
     fn eq(&self, other: &Self) -> bool {
-        self.value.eq(&other.value) && self.unit.eq(&other.unit)
+        self.value == other.value && self.unit == other.unit
     }
 }
 
 impl Eq for Number {}
 
 impl PartialOrd for Number {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-        Some(self.value.to_bits().cmp(&other.value.to_bits()))
+        if self.unit == other.unit {
+            Some(self.cmp(other))
+        } else {
+            None
+        }
     }
 }
 
 impl Ord for Number {
     fn cmp(&self, other: &Self) -> Ordering {
-        self.value.to_bits().cmp(&other.value.to_bits())
+        if self.value < other.value {
```

Comment:
Similar to the above comment. These comparisons are actually using `PartialOrd`.